controllers/chat/sendSignedMessage.js

```diff
@@ -101,18 +104,16 @@ const sendSignedMesage = async (req, res) => {
       if (isBlocked) {
         return res.status(403).json(responseError('This user does not want to receive messages'));
       }
-      // if (blockedIds.length > 0) {
-      //   for (const member of better_channel_member) {
-      //     if (member.user_id !== req.userId && blockedIds.includes(member.user_id)) {
-      //       return res
-      //         .status(403)
-      //         .json(responseError('This user does not want to receive messages'));
-      //     }
-      //   }
-      // }
+
+      channelMember = better_channel_member.filter(
+        (element) => element.user_id !== req.userId && !blockedIdsSet.has(element.user_id)
+      );
+
+      senderInfo = better_channel_member.find((element) => element.user_id === req.userId);
     }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code filters out blocked users from the `better_channel_member` array and finds the sender's information. However, there's no error handling for when `senderInfo` is `undefined`, which can happen if `req.userId` doesn't match any user ID in `better_channel_member`. This could potentially cause issues down the line when trying to access properties of `senderInfo`.



<!-- This is an auto-generated comment by OSS CodeRabbit -->